data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/utils/IdentifierShortener.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.rds.utils;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+public class IdentifierShortener {
+
+    public static String shortenIdentifier(final String identifier, final int maxLength) {
+        if (identifier.length() <= maxLength) {
+            return identifier;
+        }
+
+        try {
+            // Create SHA-256 hash
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            byte[] encodedhash = digest.digest(identifier.getBytes(StandardCharsets.UTF_8));
+
+            // Convert byte array to Base64 string
+            String base64Hash = Base64.getUrlEncoder().withoutPadding().encodeToString(encodedhash);
+
+            // Return the first maxLength characters
+            return base64Hash.substring(0, Math.min(base64Hash.length(), maxLength));
```

Comment:
QUES: what is the impact if there is a collision in the encoded result?

Reply:
There's a slim chance of collision between two pipelines configured with the same `s3_bucket` and `s3_prefix` in RDS source. In that case, the two pipelines will share the same s3 buffer, so one pipeline can process data that is supposed to be processed by the other pipeline and vice versa.

Reply:
When the bucket is used by a single user, the collision only happens within a single user's ownership. But if the bucket is multi-tenant, we will need to be more careful to avoid collisions.